cmake/pcl_find_sse.cmake

```diff
@@ -1,210 +1,257 @@
-            a = _mm_loadu_ps (vals);
-            b = a;
-            b = _mm_add_ps (a,b);
-            _mm_storeu_ps (vals,b);
-            return (0);
-        }"
-        HAVE_SSE_EXTENSIONS)
-
-    set(CMAKE_REQUIRED_FLAGS)
+      #include <xmmintrin.h>
+      int main ()
+      {
+          __m128 a, b;
+          float vals[4] = {0};
+          a = _mm_loadu_ps (vals);
+          b = a;
+          b = _mm_add_ps (a,b);
+          _mm_storeu_ps (vals,b);
+          return (0);
+      }"
+      HAVE_SSE_EXTENSIONS)
 
-    if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
-        if(HAVE_SSE4_2_EXTENSIONS)
-            list(APPEND SSE_FLAGS "-msse4.2" "-mfpmath=sse")
-        elseif(HAVE_SSE4_1_EXTENSIONS)
-            list(APPEND SSE_FLAGS "-msse4.1" "-mfpmath=sse")
-        elseif(HAVE_SSSE3_EXTENSIONS)
-            list(APPEND SSE_FLAGS "-mssse3" "-mfpmath=sse")
-        elseif(HAVE_SSE3_EXTENSIONS)
-            list(APPEND SSE_FLAGS "-msse3" "-mfpmath=sse")
-        elseif(HAVE_SSE2_EXTENSIONS)
-            list(APPEND SSE_FLAGS "-msse2" "-mfpmath=sse")
-        elseif(HAVE_SSE_EXTENSIONS)
-            list(APPEND SSE_FLAGS "-msse" "-mfpmath=sse")
-        else()
-            # Setting -ffloat-store to alleviate 32bit vs 64bit discrepancies on non-SSE
-            # platforms.
-            list(APPEND SSE_FLAGS "-ffloat-store")
-        endif()
-    elseif(MSVC AND NOT CMAKE_CL_64)
-        if(HAVE_SSE2_EXTENSIONS)
-            list(APPEND SSE_FLAGS "/arch:SSE2")
-        elseif(HAVE_SSE_EXTENSIONS)
-            list(APPEND SSE_FLAGS "/arch:SSE")
-        endif()
-    endif()
-
-    if(MSVC)
-        if(HAVE_SSSE3_EXTENSIONS)
-            string(APPEND SSE_DEFINITIONS " -D__SSSE3__")
-        endif()
-        if(HAVE_SSE2_EXTENSIONS)
-            string(APPEND SSE_DEFINITIONS " -D__SSE2__")
-        endif()
-        if(HAVE_SSE_EXTENSIONS)
-            string(APPEND SSE_DEFINITIONS " -D__SSE__")
-        endif()
-    endif()
-    string(REPLACE ";" " " SSE_FLAGS_STR "${SSE_FLAGS}")
-endmacro()
+    if(HAVE_SSE_EXTENSIONS)
+      set(SSE_LEVEL 1.0)
+    endif()
+  endif()
+  
+  if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
+    if(SSE_LEVEL GREATER_EQUAL 1.0)
+      if(SSE_LEVEL GREATER_EQUAL 4.2)
+        set(SSE_FLAGS "-msse4.2")
+      elseif(SSE_LEVEL GREATER_EQUAL 4.1)
+        set(SSE_FLAGS "-msse4.1")
+      elseif(SSE_LEVEL GREATER_EQUAL 3.1)
+        set(SSE_FLAGS "-msse3")
+      elseif(SSE_LEVEL GREATER_EQUAL 3.0)
+        set(SSE_FLAGS "-msse3")
+      elseif(SSE_LEVEL GREATER_EQUAL 2.0)
+        set(SSE_FLAGS "-msse2")
+      else()
+        set(SSE_FLAGS "-msse")
+      endif()
+      string(APPEND SSE_FLAGS " -mfpmath=sse")
+    else()
+      # Setting -ffloat-store to alleviate 32bit vs 64bit discrepancies on non-SSE
+      # platforms.
+      list(APPEND SSE_FLAGS "-ffloat-store")
+    endif()
+  elseif(MSVC AND NOT CMAKE_CL_64)
```

Comment:
Just saw in latest [documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_CL_64.html), that `CMAKE_CL_64` is deprecated and the alternate is already supported by CMake 3.5.

Reply:
Ahh, I'll update it later.